### ISSUE TITLE
Sending '--cmd quit' doesn't immediately stop twinkle

### ISSUE THREAD

**Issue by wurtel2** [LubosD/twinkle#143]
If I start "twinkle -c whatever" in one window, I can tell it to dial via "--call number", and "--cmd bye" will disconnect the call. If, however, I want to stop twinkle completely, this can't be done via the socket, as that will set the end_interface variable, but that is only evaluated after a line has been entered in the interactive console, which makes the "--cmd quit" command quite useless.

It would be good if "--cmd quit" immediately starts the shutdown of the session.

**Comment by fbriere** [comment]
I haven't checked the code yet, but `--cmd quit` on its own seems to be working fine for me.  You're saying this is not the case for you?

**Comment by wurtel2** [comment]
It is not.
I have to say that I am using the Debian 1.10.1 version.

It appears to me that the interactive thread is still waiting for readline() in this loop:

    ```
   while (!end_interface) {
               char *command_line = tw_readline(CLI_PROMPT);
               if (!command_line){
                       cout << endl;
                       break;
               }

               exec_command(command_line);
       }
```
(around line 2210 in userintf.cpp).

The socket thread has set end_interface to 1, but until tw_readline() here returns, that has no effect.


I will try to get 1.10.2 compiled from source to see if that makes a difference.

**Comment by fbriere** [comment]
AFAIK, this bit of code only applies to `twinkle-console`, not the GUI version.  Unless I misunderstood you and you were talking about the console version all along?  (I don't think you can even send commands to `twinkle-console`, but I just wanted to make sure.)

**Comment by wurtel2** [comment]
Yes, the "-c" option is for CLI mode.

I want to use twinkle to make calls when an alarm input is triggered, in addition to an SMS being sent.
It basically works, but I need to be able to stop twinkle afterwards so I can also stop pulseaudio which otherwise eats 50% of my cpu cycles passing silence around. I use pulseaudio to fake audio input into twinkle, which works fine, although having an option to send the contents of a WAV files would be wonderful...

**Comment by fbriere** [comment]
My bad, I did not realize that the `-c` option does not invoke the same code as `twinkle-console` (even though they look eerily similar), and that the former does support socket commands, unlike the latter.  Indeed, I see the same behavior as you described, and your analysis seems spot on.

My apologies for the confusion.

**Comment by fbriere** [comment]
@wurtel2 I have just submitted PR #144 to fix this bug.  Enjoy!  :smile:

(And regarding the ability to send audio from a file, feel free to open a new issue requesting such a feature.)

**Comment by wurtel2** [comment]
Thanks for the quick response! However it'll be next weekend before I have time to try it. I'll let you know.

**Comment by wurtel2** [comment]
I finally had time this weekend.
It seems to work just fine! Thank you for your work. I hope this gets merged soon.

**Comment by fbriere** [comment]
Thank you for your feedback!

(No need to close the issue; this will be done automatically once #144 is merged.)
